src/python/pants/backend/project_info/tasks/dependencies.py

```diff
@@ -23,39 +23,53 @@ def _is_jvm(target):
   @classmethod
   def register_options(cls, register):
     super().register_options(register)
-    register('--internal-only', type=bool,
-             help='Specifies that only internal dependencies should be included in the graph '
-                  'output (no external jars).')
+    register(
+      '--type', type=DependencyType, default=DependencyType.INTERNAL,
+      help="Which types of dependencies to find, where `internal` means source code dependencies "
+           "and `external` means 3rd party requirements and JARs."
+    )
+    register(
+      '--internal-only', type=bool,
+      help='Specifies that only internal dependencies should be included in the graph output '
+           '(no external jars).',
+      removal_version="1.27.0.dev0",
+      removal_hint="Use `--dependencies-type=internal` instead.",
+    )
     register(
       '--external-only',
       type=bool,
-      help='Specifies that only external dependencies should be included in the graph output (only external jars).',
-      removal_version="1.26.0.dev0",
-      removal_hint="This feature is being removed. If you depend on this functionality, please let us know in the "
-                   "#general channel on Slack at https://pantsbuild.slack.com/. \nYou can join the pants slack here: "
-                   "https://pantsslack.herokuapp.com/ ",
+      help='Specifies that only external dependencies should be included in the graph output '
+           '(only external jars).',
+      removal_version="1.27.0.dev0",
+      removal_hint="Use `--dependencies-type=external` instead.",
     )
 
   def __init__(self, *args, **kwargs):
     super().__init__(*args, **kwargs)
-
-    self.is_internal_only = self.get_options().internal_only
-    self.is_external_only = self.get_options().external_only
-    if self.is_internal_only and self.is_external_only:
-      raise TaskError('At most one of --internal-only or --external-only can be selected.')
+    opts = self.get_options()
+    type_configured = not opts.is_default("type")
+    if type_configured:
+      self.dependency_type = opts.type
+      return
+    else:
+      if opts.internal_only and opts.external_only:
+        raise TaskError('At most one of --internal-only or --external-only can be selected.')
+      if opts.internal_only:
+        self.dependency_type = DependencyType.INTERNAL
+      elif opts.external_only:
+        self.dependency_type = DependencyType.EXTERNAL
+      else:
+        self.dependency_type = DependencyType.INTERNAL_AND_EXTERNAL
 
   def console_output(self, unused_method_argument):
+    opts = self.get_options()
     deprecated_conditional(
-      lambda: not self.is_external_only and not self.is_internal_only,
-      removal_version="1.26.0.dev0",
+      lambda: opts.is_default("type") and not opts.internal_only and not opts.external_only,
+      removal_version="1.27.0.dev0",
```

Comment:
I bumped these up one major version because Stu mentioned in a different PR that that's a good practice when we deprecate the prior solution to a deprecation that's still in flight.